pkg/emailtrigger/emailtrigger.go

```diff
@@ -0,0 +1,132 @@
+package emailtrigger
```

Comment:
I refactored the code a bit to be able to share some common trigger code from both smtp and sendgrid.